xarray/conventions.py

```diff
@@ -82,7 +82,8 @@ def maybe_encode_nonstring_dtype(var, name=None):
         if dtype != var.dtype:
             if np.issubdtype(dtype, np.integer):
                 if (np.issubdtype(var.dtype, np.floating) and
-                        '_FillValue' not in var.attrs):
+                    '_FillValue' not in var.attrs and
+                    'missing_value' not in var.attrs):  # noqa
```

Comment:
I think you can avoid "noqa" here if you add an extra level of indentation on these two lines.

Reply:
Ah thanks. That passes flake8 locally.